pkg/controller/storagecluster/storagecluster.go

```diff
@@ -1236,30 +1236,40 @@ func getDefaultStorageNodesDisaggregatedMode(
 
 // getDefaultMaxStorageNodesPerZone aims to return a good value for MaxStorageNodesPerZone with the
 // intention of having at least 3 nodes in the cluster.
-func getDefaultMaxStorageNodesPerZone(
+func (c *Controller) getDefaultMaxStorageNodesPerZone(
 	nodeList *v1.NodeList,
 	cluster *corev1.StorageCluster,
 	recorder record.EventRecorder,
 ) (uint32, error) {
 	if len(nodeList.Items) == 0 {
 		return 0, nil
 	}
+	filteredList := v1.NodeList{}
+	for _, node := range nodeList.Items {
+		shouldRun, _, err := c.nodeShouldRunStoragePod(&node, cluster)
```

Comment:
are we not considering should continue running here? 
also @haibinxie I moved those functions from controller to portworx API, calling controller util function here might be hard to resolve.. do we have a better way for this? like adding additional conditions after `k8s.CheckPredicatesForStoragePod`? I feel this `nodeShouldRunStoragePod` is too complicated.

Reply:
shouldContinue will return true for master nodes. So not using it.